File: templates/go/internal/modules/health/endpoint/http.go
package endpoint

import (
	"net/http"

	"example.com/myapp/internal/modules/health/feature/check"
)

// Register wires the health module's HTTP routes.
// Each module owns its route registration — vertical slice pattern.
func Register(mux *http.ServeMux) {
	svc := check.NewService()
	handler := check.NewHandler(svc)

	mux.HandleFunc("GET /health", handler.Handle)
}

File: templates/python/app/modules/health/endpoint/http.py
"""Health module HTTP endpoint registration."""


def register_routes(app) -> None:
    """Register health check routes with the web framework."""
    from app.modules.health.feature.check import HealthService, HealthHandler

    service = HealthService()
    handler = HealthHandler(service)

    # Framework-specific route registration goes here
    # Example for Flask: app.add_url_rule('/health', view_func=handler.handle)
    pass
